### PR TITLE
#169817812 user signup on barefoot nomad

### PR DESCRIPTION
**Pull request by VedaDom**
#### What does this PR do?
- Users should be able Signup/Register to Barefoot Nomad.

#### Description of Task to be completed?
- The sign up form should present to the user **descriptive** validation errors incase of any.
- The user should receive verification link on email upon successful registration.

#### How should this be manually tested?
After clone repository, go to the root of the project open path into command run the `npm install` and once setup is done click `npm start`, this will start a local server which will then automatically open your default browser, then navigate to `http://localhost:8080/auth/signup` and fill sign up form and click on sign up button.

#### What are the relevant pivotal tracker stories?
- [#169817812](https://www.pivotaltracker.com/story/show/169817812)

#### Screenshots (if appropriate)
Screen Shot 2020-02-07 at 08 16 26

### CODE REVIEW

**Review by shemaeric1234 — APPROVED** [review]

**Review by mujohn26 — APPROVED** [review]
Good Work @VedaDom

**Comment by Ruqoyah** [comment]
Kindly fix the failing test on CI.

File: src/components/signup/signup_form.component.jsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+                                                </svg>
+                                            </SvgIcon>}>
+                                        Sign up with Google
+                                </Button>
+                                </Link>
+                            </Box>
+                        </Grid>
+                        <Grid item xs={6}>
+                            <Box pl={1}>
+                                <Link style={{ textDecoration: 'none' }} href="https://blackninjas-backend-staging.herokuapp.com/api/v1/auth/facebook">
```
Comment (bdushimi) [inline src/components/signup/signup_form.component.jsx]: @VedaDom kindly use the env variables here too.

File: src/components/signup/signup_form.component.jsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                        <Box pt={2} pb={2}>
+                            <Typography variant="body2">
+                                Or
+                            </Typography>
+                        </Box>
+                    </Grid>
+                    <Grid justify="space-between" container>
+                        <Grid item xs={6}>
+                            <Box pr={1}>
+                                <Link style={{ textDecoration: 'none' }} href="https://blackninjas-backend-staging.herokuapp.com/api/v1/auth/google">
```
Comment (bdushimi) [inline src/components/signup/signup_form.component.jsx]: @VedaDom kindly use env. variables here too

**Review by bdushimi — CHANGES_REQUESTED** [review]

**Review by bdushimi — APPROVED** [review]

**Pull request merged**